### PR TITLE
Retry for EINTR (Fix for #16)

### PR DESCRIPTION
**Pull request by j16sdiz**

### CODE REVIEW

**Comment by j16sdiz** [comment]
ref #16

File: src/librabbitmq/amqp_private.h [hunk 1]
```diff
@@ -125,6 +125,14 @@ extern int amqp_encode_table(amqp_bytes_t encoded,
     _result;					\
   })
 
+#define AMQP_CHECK_WRITE_RESULT(expr)			\
+  ({						\
+    int _result;            \
+    do {_result = (expr);} while (_result == -EINTR);			\
```
Comment (postwait) [inline src/librabbitmq/amqp_private.h]: write(2) will return -1 on error and set errno to EINTR...

I believe that should be

```
while(-1 == (_result = expr)) && errno == EINTR);
```

File: src/librabbitmq/amqp_private.h [hunk 2]
```diff
@@ -125,6 +125,14 @@ extern int amqp_encode_table(amqp_bytes_t encoded,
...
+    if (_result < 0) return _result;		\
```
Comment (postwait) [inline src/librabbitmq/amqp_private.h]: This line is unnecessary as you return result on the next line regardless.

**Comment by keithf4** [comment]
This is also changing the source of an external library we don't maintain

https://github.com/alanxz/rabbitmq-c

Be curious if this issue is handled better in a newer version of that library and we'd be better off trying to see if that could be incorporated. Would also possibly fix another open issue https://github.com/omniti-labs/pg_amqp/issues/5

**Comment by j16sdiz** [comment]
@keithf4

> This is also changing the source of an external library we don't maintain
> https://github.com/alanxz/rabbitmq-c

I have checked @alanxz 's version before my change -- he have that fixed already.
but they have too many different that may need to rewrite those glue.

**Comment by j16sdiz** [comment]
I have another branch https://github.com/j16sdiz/pg_amqp/tree/librabbitmq-0.4 that use 0.4 from upstream.
It is free of compile error, but I am not confident it would work correctly.

**Pull request closed without merging**